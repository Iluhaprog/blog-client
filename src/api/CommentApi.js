import api from './api';

function create(comment) {
    return api.post('/comment/create', { comment });
}

function getById(id) {
    return api.get('/comment/getById', {
        params: { id }
    });
}

function getAll(page, limit) {
    return api.get(`/comment/getAll/${page}/${limit}`);
}

function getByPostId(postId) {
    return api.get('/comment/getByPostId', {
        params: { postId }
    });
}

function getByUserId(userId) {
    return api.get('/comment/getByUserId', {
        params: { userId }
    });
}

function update(comment) {
    return api.put('/comment/update', { comment });
}

function deleteById(id) {
    return api.delete('/comment/deleteById', {
        params: { id }
    });
}

export default {
    create,
    getById,
    getAll,
    getByPostId,
    getByUserId,
    update,
    deleteById,
};