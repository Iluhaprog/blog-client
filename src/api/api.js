import axios from 'axios';

const BASE_URL = process.env.API_URL;

const api = axios.create({
    baseURL: BASE_URL,
    withCredentials: true,
});

export default api;