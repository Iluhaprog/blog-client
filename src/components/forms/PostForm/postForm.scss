@import '../../../vars.scss';

.post-form {
    max-width: 60%;
    width: 100%;

    .drop-zone {
        min-width: 120px;
        margin-right: 20px;
    }

    .form__box {
        max-width: 100%;
    }

    .send-button {
        margin-top: 20px;
    }

    .success-button {
        font-size: 20px;
        margin-right: 10px;
        flex: 0 0 auto;
    }

    .toggler-title {
        margin: 0;
        margin-right: 15px;
        font-weight: 300;
        color: $text-color-5;
        transition: .2s ease-in-out;
        margin-bottom: 20px;

        &__visible {
            color: $text-color-6;
        }
    }

    &__tags {
        min-width: 110px;

        .button {
            margin: 0 auto;
            margin-top: 10px;
        }

        p {
            margin: 0;
            padding: 10px;
            text-align: center;
            color: $text-color-6;
            border: $border-2;
            border-radius: 5px 0 0 5px;
        }

    }

    .multiple-select {
        border-radius: 0 5px 0 0;
    }
}