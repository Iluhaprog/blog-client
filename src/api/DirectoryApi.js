import api from './api';

export const GET_DIRECTORY_BY_ID = '/directory/getById';

function getById(id) {
    return api.get(GET_DIRECTORY_BY_ID, {
        params: {
            id,
        },
    });
}

export default {
    getById,
};