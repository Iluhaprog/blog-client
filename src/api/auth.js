import { api } from './api';
import {requestWithToken} from "../utils/decorators/decorators";
import {base64Encode} from "../utils/base64";

export function login(username, password) {
    return api.post('/auth/login', null, {
        withCredentials: true,
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${base64Encode(`${username}:${password}`)}`,
        },
    });
}

export function refreshToken() {
    return requestWithToken(
        (headers) => api.get('/auth/refresh-token', {
            headers,
            params: {
                token: localStorage.getItem(process.env.REACT_APP_REFRESH_TOKEN_KEY),
            },
        }),
    );
}

export function logout() {
    return requestWithToken(
        (headers) => api.get('/auth/logout', {
            headers,
            params: {
                token: localStorage.getItem(process.env.REACT_APP_REFRESH_TOKEN_KEY),
            },
        }),
    );
}

export function logoutAll() {
    return requestWithToken(
        (headers) => api.get('/auth/logoutAll', {
            headers
        }),
    );
}