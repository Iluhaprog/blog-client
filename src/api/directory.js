import { api } from './api';
import {requestWithToken} from "../utils/decorators/decorators";

const DIR_URL = '/directory';

export function getAll(page, limit = process.env.REACT_APP_PAGINATION_LIMIT) {
    return api.get(DIR_URL, {
        params: { page, limit },
    });
}

export function create(dir) {
    return requestWithToken(
        (headers) => api.post(DIR_URL, dir, {
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                ...headers,
            }
        }),
    );
}

export function remove(id) {
    return requestWithToken(
        (headers) => api.delete(DIR_URL,{
            params: { id },
            headers,
        }),
    );
}