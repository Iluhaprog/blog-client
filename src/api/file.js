import {api} from './api';
import {requestWithToken} from '../utils/decorators/decorators';
import {Filter} from './filters';

const FILE_URL = '/file';

/**
 * @param {number} page
 * @param {number} limit
 * @param {string} order
 * @return {Promise<AxiosResponse<any>>}
 */
export function getAll(
    page,
    limit = process.env.REACT_APP_PAGINATION_LIMIT,
    order = Filter.DESC,
) {
  return api.get(FILE_URL, {
    params: {page, limit, order},
  });
}

/**
 * @param {number} dirId
 * @param {string} order
 * @return {Promise<AxiosResponse<any>>}
 */
export function getByDirId(dirId, order = Filter.DESC) {
  return api.get(`${FILE_URL}/by-dir`, {
    params: {
      dirId,
      order,
    },
  });
}

/**
 * @param {FormData} formData
 * @param {number} dirId
 * @return {Promise<AxiosResponse<any>>}
 */
export function create(formData, dirId) {
  return requestWithToken(
      (headers) => api.post(FILE_URL, formData, {
        params: {dir: dirId},
        headers,
      }),
  );
}

/**
 * @param {number} id
 * @return {Promise<AxiosResponse<any>>}
 */
export function remove(id) {
  return requestWithToken(
      (headers) => api.delete(FILE_URL, {
        params: {id},
        headers,
      }),
  );
}
