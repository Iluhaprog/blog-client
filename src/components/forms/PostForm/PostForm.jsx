import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field, change, arrayPush, getFormValues } from 'redux-form';
import { SendButton, SuccessButton } from '../../buttons';
import { Row } from '../../containers';
import { 
    LabeledInput, 
    MarkdownRedactor, 
    DropZone, 
    MultipleSelect,
    Textarea,
    LabeledToggler
} from '../../formElements';
import { addTagFetch, getAllFetch } from '../../../actoins/tag';
import { setModal } from '../../../actoins/modal'
import './postForm.scss';

function PostForm(props) {
    const { handleSubmit, openModalWindow, setSelect, setVisible } = props;

    useEffect(() => {
        props.getAllTags();
    }, [props.tags])

    return (
        <form onSubmit={handleSubmit} className='post-form'>
            <LabeledInput 
                type='text'
                name='title'
                component='input'
                label='Title'
            />
            <Row>
                <Field
                    name='preview'
                    component={DropZone}
                />
                <Textarea 
                    name='description'
                    placeholder='Description'
                />
            </Row>
            <MarkdownRedactor 
                name='text'
                placeholder='Text'
                value={props.formValues ? props.formValues.text : ''}
            />
            <LabeledToggler 
                value={props.formValues ? props.formValues.visible : false}
                handleChange={setVisible}
                label='Visible: '
            />
            <Row alignItems='fs'>
                <div className='post-form__tags'>
                    <p>Tags:</p>
                    <SuccessButton 
                        text='New tag'
                        onClick={e => {
                            e.preventDefault();
                            openModalWindow();
                        }}
                    />
                </div>
                <MultipleSelect
                    values={props.selected.Tags || []}
                    options={props.tags}
                    onChange={option => setSelect(option)}
                />
            </Row>
            <SendButton 
                text='Send'
            />
        </form>
    );
}

const mapStateToProps = state => ({
    initialValues: {
        ...state.post.selected,
    },
    selected: state.post.selected,
    tags: state.tags.tags,
    formValues: getFormValues('postForm')(state),
});

const mapDispatchToProps = dispatch => ({
    openModalWindow: () => {
        dispatch(setModal({
            text: 'Write tag title:',
            visible: true,
            withInput: true,
            handleSuccess: (title, fail) => {
                dispatch(addTagFetch({ title }))
            }
        }));
    },
    addTag: tag => {
        dispatch(addTagFetch(tag));
    },
    getAllTags: () => {
        dispatch(getAllFetch())
    },
    setSelect: value => {
        dispatch(arrayPush('postForm', 'Tags', value))
    },
    setVisible: value => {
        dispatch(change('postForm', 'visible', value))
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
    form: 'postForm',
    enableReinitialize: true,
})(PostForm));