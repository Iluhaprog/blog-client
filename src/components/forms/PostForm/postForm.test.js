import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'
import MockAdapter from 'axios-mock-adapter';
import thunk from 'redux-thunk'
import { render, fireEvent, waitFor } from '@testing-library/react';
import PostForm from './PostForm';
import api from '../../../api/api';

const initPostState = {
    post: {
        selected: {
            title: '',
            description: '',
            text: '',
            preview: '',
            visible: false,
            date: '',
            directoryId: 0,
            Tags: [],
        },
        files: [],
        array: [],
        total: 0,
    },
    tags: {
        tags: [],
    },
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('Test view of post form', () => {
    const store = mockStore(initPostState);
    const { asFragment } = render(
        <Provider store={store}>
            <PostForm onSubmit={() => {}} />)
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});