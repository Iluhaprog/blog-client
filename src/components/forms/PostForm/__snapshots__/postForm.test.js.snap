// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of post form 1`] = `
<DocumentFragment>
  <form
    class="post-form"
  >
    <div
      class="form__box row-reverse"
    >
      <input
        name="title"
        type="text"
        value=""
      />
      <label
        for="title"
      >
        Title
      </label>
    </div>
    <div
      class="row   "
    >
      <div
        class="drop-zone "
      >
        <img
          alt="empty"
          class="empty"
          src="picture.svg"
        />
        <input
          type="file"
        />
      </div>
      <textarea
        class="textarea"
        name="description"
        placeholder="Description"
      />
    </div>
    <div
      class="markdown-redactor "
    >
      <div
        class="markdown-redactor__write-zone"
      >
        <button
          class="button default-button success-button"
        >
          view
        </button>
        <textarea
          class="textarea"
          name="text"
          placeholder="Text"
        />
      </div>
      <div
        class="markdown-redactor__view-zone markdown-body"
      />
    </div>
    <div
      class="row   "
    >
      <p
        class="toggler-title 
                            "
      >
        Visible:
      </p>
      <div
        class="toggler"
      >
        <div
          class="toggler-box"
        >
          <div
            class="toggler-box__item "
          />
        </div>
      </div>
    </div>
    <div
      class="row  row_ai-fs "
    >
      <div
        class="post-form__tags"
      >
        <p>
          Tags:
        </p>
        <button
          class="button default-button success-button"
        >
          New tag
        </button>
      </div>
      <div
        class="multiple-select"
        tabindex="0"
      >
        <ul
          class="values"
        >
          <div
            class="row row_jc-fs row_ai-c row_w-w"
          />
        </ul>
        <div
          class="options"
        />
      </div>
    </div>
    <button
      class="button send-button"
    >
      Send
    </button>
  </form>
  )
</DocumentFragment>
`;
